src/main/java/org/springframework/data/elasticsearch/core/mapping/IndexInformation.java

```diff
@@ -0,0 +1,120 @@
+import org.elasticsearch.cluster.metadata.AliasMetadata;
+import org.elasticsearch.cluster.metadata.MappingMetadata;
+import org.elasticsearch.common.settings.Settings;
+
+/**
+ * Immutable object that holds information(name, settings, mappings, aliases) about an Index
+ *
+ * @author George Popides
+ * @since 4.2
+ */
+public class IndexInformation {
+	private final String name;
+	private final Settings settings;
+	private final MappingMetadata mappings;
+	private final List<AliasMetadata> aliases;
+
+
+	public static List<IndexInformation> createList(GetIndexResponse getIndexResponse) {
+		return buildIndexInformationList(getIndexResponse);
+	}
+
+	public static List<IndexInformation> createList(org.elasticsearch.action.admin.indices.get.GetIndexResponse getIndexResponse) {
+		return buildIndexInformationList(getIndexResponse);
+	}
+
+	private static List<IndexInformation> buildIndexInformationList(GetIndexResponse response) {
+		List<IndexInformation> indexInformationList = new ArrayList<>();
+		for (String indexName : response.getIndices()) {
+			indexInformationList.add(IndexInformation.of(indexName, response));
+		}
+		return indexInformationList;
+	}
+
+	private static List<IndexInformation> buildIndexInformationList(org.elasticsearch.action.admin.indices.get.GetIndexResponse response) {
+		List<IndexInformation> indexInformationList = new ArrayList<>();
+		for (String indexName : response.getIndices()) {
+			indexInformationList.add(IndexInformation.of(indexName, response));
+		}
+		return indexInformationList;
+	}
+
+	private static IndexInformation of(String indexName, org.elasticsearch.action.admin.indices.get.GetIndexResponse getIndexResponse) {
+		Settings settings = getIndexResponse.getSettings().get(indexName);
+		MappingMetadata mappingMetadata = getIndexResponse.getMappings().get(indexName).get("indexName");
+		List<AliasMetadata> aliases = getIndexResponse.getAliases().get(indexName);
+
+		return new IndexInformation(indexName, settings, mappingMetadata, aliases);
+	}
+
+	private static IndexInformation of(String indexName, GetIndexResponse getIndexResponse) {
+		Settings settings = getIndexResponse.getSettings().get(indexName);
+		MappingMetadata mappingMetadata = getIndexResponse.getMappings().get(indexName);
+		List<AliasMetadata> aliases = getIndexResponse.getAliases().get(indexName);
+
+		return new IndexInformation(indexName, settings, mappingMetadata, aliases);
+	}
+
+	public String getName() {
+		return name;
+	}
+
+	public Settings getSettings() {
+		return settings;
+	}
+
+	public List<AliasMetadata> getAliases() {
+		return aliases;
+	}
+
+	public MappingMetadata getMappings() {
+		return mappings;
+	}
+
+	private IndexInformation(String name, Settings settings, MappingMetadata mappingMetadata, List<AliasMetadata> aliases) {
+		this.name = name;
+		this.settings = settings;
+		this.mappings = mappingMetadata;
+		this.aliases = aliases;
+	}
+
+	@Override
+	public String toString() {
```

Comment:
If we need a `toString()` it should contain the content of the properties as well.